sql/log.cc

```diff
@@ -710,10 +714,22 @@ bool File_query_log::write_slow(THD *thd, ulonglong current_utime,
     /* Note that my_b_write() assumes it knows the length for this */
     if (my_b_write(&log_file, (uchar *)buff, buff_len)) goto err;
 
+    /* We could rewrite below to avoid code duplication. */
+    /* This could allow a single my_b_printf, but would prevent code grepping. */
+    /* So we accept code duplication. */
     buff_len = snprintf(buff, 32, "%5u", thd->thread_id());
-    if (my_b_printf(&log_file, "# User@Host: %s  Id: %s\n", user_host, buff) ==
-        (uint)-1)
-      goto err;
+    if (!opt_log_slow_extra_db) {
+      if (my_b_printf(&log_file, "# User@Host: %s  Id: %s\n", user_host, buff) == (uint)-1)
+        goto err;
+    } else {
+      log_db_change = false;  /* Do not log db change if db is in the comment line. */
+      db[0] = 0;  /* Reset db triggers logging db change after disabling log_slow_extra_db. */
+      if (thd->db().str) {
```

Comment:
Honestly I'd just do this bit and not worry about the extra system variable.

adjust scripts/mysqldumpslow.pl.in

When finished add a sample: https://github.com/percona/percona-toolkit/tree/3.x/t/lib/samples/slowlogs

regex:
https://github.com/percona/percona-toolkit/blob/83ba470afe5008276a7656102b8abe0cf40a31e6/lib/SlowLogParser.pm#L45

Reply:
Thanks for the feedback Daniel.

> I'd just do this bit and not worry about the extra system variable.

If I do this, I might break script parsing the slow log, which I would like to avoid.  With the variable and a default of ON, people will see the brreakage and they will have the liberty to fix their script of disable the feature.  I prefer to give them this libeerty.  Also, with a variable, this patch can be back-ported in 8.0 and 8.4, with a default of OFF.  So I think I will keep it.

> adjust scripts/mysqldumpslow.pl.in

No need, it already swallows everything after `Id: ...`.
- https://github.com/jfg956/mysql-server/blob/mysql-8.4.0/scripts/mysqldumpslow.pl.in#L104

> When finished add a sample [...]

Thanks for bringing this to my attention, I will look into this.

> regex

Thanks for the pointer.  Cheers, J-F

Reply:
Percona notified: https://perconadev.atlassian.net/browse/PT-2370